### ISSUE TITLE
Training limit understanding

### ISSUE THREAD

**Issue by fernaper** [alteryx/evalml#2390]
Hello,

I know Evalml has an early stopping system but I don't understand why doing the same things with Evalml and with MLJAR, MLJAR obtains a better accuracy (with much more time).

Almost all the problems I tried with Evalml stopped training in less than a minute, this is crazy, but for production models, I prefer a better accuracy against speed.

Is there any way to perform better by taking more time?

You can check here the comparation between both frameworks in time and performance with the same Dataset: [Comparation.zip](https://github.com/alteryx/evalml/files/6670089/Comparation.zip)

Thank you very much for everything.
I really prefer to use Evalml but I cannot understand why it stops so quickly.

**Comment by jeremyliweishih** [comment]
Hey @fernaper! I'm taking a look now - will get back to you shortly.

**Comment by jeremyliweishih** [comment]
@fernaper a couple things:

1. Your original experiment ends early because `max_batches` is set to 1 by default in AutoMLSearch so the search will only last 1 batch of pipelines. EvalML has early stopping turned off by default so that isn't the culprit either. There are a couple parameters for stopping criterion in `AutoMLSearch`. One way is to set the max_batches parameter which controls the maximum number of rounds of AutoML to evaluate, where each round may train and score a variable number of pipelines. Another way is to set the max_iterations parameter which controls the maximum number of candidate models to be evaluated during AutoML. By default, AutoML will search for a single batch. In the edited notebook attached, I include an example of using `max_time` to match your mljar notebook.

2. It looks like mljar in “competition” mode has ensembling enabled. To do the same in EvalML you would need to set `ensembling = True` in `AutoMLSearch`. Ensembling will be run in a separate batch after every allowed pipeline class has been iterated over.

3. `evalml.preprocessing.split_data` uses sklearn’s `ShuffleSplit` and in your example for mjlar you use sklearn’s `train_test_split`. I believe the underlying implementation of these methods are the same but `random_state` should be set for `train_test_split` to match `random_seed in `evalml.preprocessing.split_data`.

4. EvalML uses an optimization metric to optimize for and select the best pipeline. For regression problems we default to R2. This can be specified using the `objective` parameter in `AutoMLSearch`. It looks like mjlar defaults to RMSE.

I’ve adjusted the following in the notebooks:
- standardize splits between both notebooks using the evalml split
- set objective in EvalML to `ExpVar` and evaluation metric in mjlar to R2 (most similar metric I could find)
- set max_time to 30 * 60 to standardize stopping criterion

In EvalML, the best performing pipeline is a CatBoost pipeline with a ExpVar holdout of 0.704 and mjlar has a holdout score of 0.708. You can use these notebooks as an example of setting stopping criterion and configuring objectives for EvalML. We’re also currently in the works of implementing a new algorithm that fits your use case of a longer running but higher performing search. Let me if you have any other questions about configuring AutoMLSearch or any other EvalML questions!
[edited_notebooks.zip](https://github.com/alteryx/evalml/files/6672137/edited_notebooks.zip)
